File: apps/android/app/src/qa/java/chat/rogi/rogichat/preview/WireframeState.kt
package chat.rogi.rogichat.preview

// UI-only state. This type must never become a session, repository, or API DTO.
enum class PreviewRole(val label: String) { FAN("팬"), STREAMER("스트리머") }
enum class PreviewPage(val title: String) {
    WELCOME("로기챗 QA"), LINK("SOOP 계정 연결"), ROOMS("대화"), CHAT("대화방"),
    SETTINGS("설정"), PROFILE("내 프로필"), ACCOUNT("계정 관리"), REPORT("신고 및 차단")
}
enum class ListScenario(val label: String) { CONTENT("목록"), LOADING("로딩"), EMPTY("빈 목록"), ERROR("오류") }
enum class PreviewAudience(val label: String) { PRIVATE("개인 답장"), SHARED("전체 대화") }
data class SampleRoom(val id: String, val title: String, val summary: String)

object WireframeFixtures {
    val rooms = listOf(
        SampleRoom("sample-room-a", "샘플 스트리머 A", "함께 나누는 오늘의 이야기"),
        SampleRoom("sample-room-b", "샘플 스트리머 B", "다음 대화를 기다리고 있어요"),
    )
    val fans = listOf("샘플 팬 A", "샘플 팬 B")
}

data class WireframeState(
    val role: PreviewRole = PreviewRole.FAN,
    val page: PreviewPage = PreviewPage.WELCOME,
    val linkPreviewPassed: Boolean = false,
    val roomId: String? = null,
    val scenario: ListScenario = ListScenario.CONTENT,
    val audience: PreviewAudience = PreviewAudience.PRIVATE,
    val target: String? = null,
    val draft: String = "",
) {
    fun switchRole(value: PreviewRole) = if (value == role) this else WireframeState(role = value)
    fun previewLink() = if (page == PreviewPage.WELCOME) copy(page = PreviewPage.LINK) else this
    fun previewRooms() = if (page == PreviewPage.LINK) copy(page = PreviewPage.ROOMS, linkPreviewPassed = true) else this
    fun openRoom(id: String): WireframeState {
        if (!linkPreviewPassed || page != PreviewPage.ROOMS || scenario != ListScenario.CONTENT ||
            WireframeFixtures.rooms.none { it.id == id }) return this
        return copy(page = PreviewPage.CHAT, roomId = id, draft = "", target = null, audience = PreviewAudience.PRIVATE)
    }
    fun open(value: PreviewPage): WireframeState = when {
        value == PreviewPage.SETTINGS && page == PreviewPage.ROOMS -> copy(page = value)
        value in listOf(PreviewPage.PROFILE, PreviewPage.ACCOUNT) && page == PreviewPage.SETTINGS -> copy(page = value)
        value == PreviewPage.REPORT && page == PreviewPage.CHAT -> copy(page = value)
        else -> this
    }
    fun changeAudience(value: PreviewAudience): WireframeState {
        if (role != PreviewRole.STREAMER || page != PreviewPage.CHAT) return this
        // A draft must never silently move to a different recipient/audience.
        return if (value == audience) this else copy(audience = value, target = null, draft = "")
    }
    fun selectTarget(value: String): WireframeState {
        if (role != PreviewRole.STREAMER || page != PreviewPage.CHAT || value !in WireframeFixtures.fans) return this
        return if (target == value && audience == PreviewAudience.PRIVATE) this
        else copy(audience = PreviewAudience.PRIVATE, target = value, draft = "")
    }
    val canCompose: Boolean get() = page == PreviewPage.CHAT && roomId != null &&
        (role == PreviewRole.FAN || audience == PreviewAudience.SHARED || target != null)
    fun editDraft(value: String) = if (canCompose) copy(draft = value.take(2000)) else this
    fun back(): WireframeState = when (page) {
        PreviewPage.WELCOME -> this
        PreviewPage.LINK, PreviewPage.ROOMS -> WireframeState(role = role)
        PreviewPage.CHAT -> copy(page = PreviewPage.ROOMS, roomId = null, draft = "", target = null)
        PreviewPage.SETTINGS -> copy(page = PreviewPage.ROOMS)
        PreviewPage.PROFILE, PreviewPage.ACCOUNT -> copy(page = PreviewPage.SETTINGS)
        PreviewPage.REPORT -> copy(page = PreviewPage.CHAT)
    }
}

File: apps/ios/Sources/QA/WireframeState.swift
#if ROGICHAT_QA
// UI-only state. Never use these values as a session, repository, or API DTO.
enum PreviewRole: String, CaseIterable {
    case fan = "팬", streamer = "스트리머"
}
enum PreviewPage: String, Hashable {
    case link = "SOOP 계정 연결", rooms = "대화", chat = "대화방", settings = "설정"
    case profile = "내 프로필", account = "계정 관리", report = "신고 및 차단"
}
enum ListScenario: String, CaseIterable {
    case content = "목록", loading = "로딩", empty = "빈 목록", error = "오류"
}
enum PreviewAudience: String, CaseIterable {
    case `private` = "개인 답장", shared = "전체 대화"
}
struct SampleRoom: Identifiable {
    let id: String
    let title: String
    let summary: String
}
enum WireframeFixtures {
    static let rooms = [
        SampleRoom(id: "sample-room-a", title: "샘플 스트리머 A", summary: "함께 나누는 오늘의 이야기"),
        SampleRoom(id: "sample-room-b", title: "샘플 스트리머 B", summary: "다음 대화를 기다리고 있어요")
    ]
    static let fans = ["샘플 팬 A", "샘플 팬 B"]
}

struct WireframeState {
    private(set) var role: PreviewRole = .fan
    private(set) var path: [PreviewPage] = []
    private(set) var linkPreviewPassed = false
    private(set) var roomID: String?
    var scenario: ListScenario = .content
    private(set) var audience: PreviewAudience = .private
    private(set) var target: String?
    private(set) var draft = ""

    var page: PreviewPage? { path.last }
    var canCompose: Bool {
        page == .chat && roomID != nil && (role == .fan || audience == .shared || target != nil)
    }
    mutating func reset() { self = WireframeState(role: role) }
    mutating func switchRole(_ value: PreviewRole) {
        if value != role { self = WireframeState(role: value) }
    }
    mutating func previewLink() {
        guard path.isEmpty else { return }
        path = [.link]
    }
    mutating func previewRooms() {
        guard page == .link else { return }
        linkPreviewPassed = true
        path = [.rooms]
    }
    mutating func openRoom(_ id: String) {
        guard linkPreviewPassed, page == .rooms, scenario == .content,
              WireframeFixtures.rooms.contains(where: { $0.id == id }) else { return }
        roomID = id
        draft = ""
        target = nil
        audience = .private
        path.append(.chat)
    }
    mutating func open(_ value: PreviewPage) {
        switch (page, value) {
        case (.rooms, .settings), (.settings, .profile), (.settings, .account), (.chat, .report):
            path.append(value)
        default: break
        }
    }
    mutating func changeAudience(_ value: PreviewAudience) {
        guard role == .streamer, page == .chat, audience != value else { return }
        audience = value
        target = nil
        draft = ""
    }
    mutating func selectTarget(_ value: String) {
        guard role == .streamer, page == .chat, WireframeFixtures.fans.contains(value) else { return }
        if target == value && audience == .private { return }
        audience = .private
        target = value
        draft = ""
    }
    mutating func editDraft(_ value: String) {
        guard canCompose else { return }
        draft = String(value.prefix(2000))
    }
    // NavigationStack back gesture/button may only remove an existing suffix.
    mutating func pop(to newPath: [PreviewPage]) {
        guard newPath.count < path.count, Array(path.prefix(newPath.count)) == newPath else { return }
        path = newPath
        if path.isEmpty { reset() }
        else if !path.contains(.chat) {
            roomID = nil
            target = nil
            draft = ""
        }
    }
}
#endif
